### ISSUE TITLE
有关训练代码的问题

### ISSUE THREAD

**Issue by SaltyFisher1** [QinYang79/RDE#8]
大佬你好，我是刚从事noisy对应性学习的研究生，您的工作非常优秀，但是我在跑您提供的开源代码时遇到一个问题，
代码    
/processor.py里的 def get_loss(model, data_loader): 在计算loss进行高斯混合之前有一句 model.eval()，这里我猜是相当于进行了模型的推断，来判断样本是否为clean样本和noisy样本不应该传梯度，但是整体模型在get_loss判断完之后，好像并没有回到 model.train()模式，但是这样也能训练，这里使我产生了困惑，期待您的解答。
![image](https://github.com/user-attachments/assets/5985ca83-257e-4e54-8570-81f4dd34adc7)

**Comment by QinYang79** [comment]
https://www.jianshu.com/p/74c02e8d28f3
这似乎确实一个疏忽之处，你可以check一下更正下该项性能是否受影响。

**Comment by QinYang79** [comment]
> https://www.jianshu.com/p/74c02e8d28f3 这似乎确实一个疏忽之处，你可以check一下更正下该项性能是否受影响。

我纠正了这一项，重新进行了实验验证，跑出来的性能会略高于论文所报告的。

**Comment by QinYang79** [comment]
> 非常感谢您的回答，可以分享一下您修改的地方吗？非常感谢

```Python
label_hat[consensus_division>1] = 1
label_hat[consensus_division<=1] = 0 

model.train()
for n_iter, batch in enumerate(train_loader):
```

**Comment by QinYang79** [comment]
⚠️ Fixing https://github.com/QinYang79/RDE/issues/8 will cause some performance degradation in noisy scenes. You can comment out "line217 model.train()" in processor/processor.py.

**Comment by ruijiecai** [comment]
杨博士您好，我是一名刚学习噪声对应学习的学生。我在阅读您的代码过程中发现一个问题，就是在processer.py里面的BetaMixture1D在代码中哪里被使用到了，他和GMM貌似没啥关系，我不能理解这个类在代码中哪里被使用了，他的作用是什么。杨博士可以解决我这个困惑么。期待杨博士的回答。

![Image](https://github.com/user-attachments/assets/b0f0eda9-c45a-49c6-9610-8680a9ee57a9)

**Comment by QinYang79** [comment]
> 杨博士您好，我是一名刚学习噪声对应学习的学生。我在阅读您的代码过程中发现一个问题，就是在processer.py里面的BetaMixture1D在代码中哪里被使用到了，他和GMM貌似没啥关系，我不能理解这个类在代码中哪里被使用了，他的作用是什么。杨博士可以解决我这个困惑么。期待杨博士的回答。
> 
> ![Image](https://github.com/user-attachments/assets/b0f0eda9-c45a-49c6-9610-8680a9ee57a9)

Sorry for the late reply. This part of the code is the code I used to partition the data using BMM. I forgot to delete it here, which does not affect RDE.

**Comment by dyt7788** [comment]
Hello Dr. Qin, I encountered a problem in reproducing the code, that is, in the case of 50% noise, I cannot reproduce the result in the paper, is there any specific parameter set?

<img width="1626" height="338" alt="Image" src="https://github.com/user-attachments/assets/263a07a6-6999-44b0-90cc-93ff0139236c" />

**Comment by QinYang79** [comment]
> > 杨博士您好，我是一名刚学习噪声对应学习的学生。我在阅读您的代码过程中发现一个问题，就是在processer.py里面的BetaMixture1D在代码中哪里被使用到了，他和GMM貌似没啥关系，我不能理解这个类在代码中哪里被使用了，他的作用是什么。杨博士可以解决我这个困惑么。期待杨博士的回答。
> > ![Image](https://github.com/user-attachments/assets/b0f0eda9-c45a-49c6-9610-8680a9ee57a9)
> 
> Sorry for the late reply. This part of the code is the code I used to partition the data using BMM. I forgot to delete it here, which does not affect RDE.

**Comment by QinYang79** [comment]
> Hello Dr. Qin, I encountered a problem in reproducing the code, that is, in the case of 50% noise, I cannot reproduce the result in the paper, is there any specific parameter set?
> 
> <img alt="Image" width="1626" height="338" src="https://private-user-images.githubusercontent.com/73678248/475912626-263a07a6-6999-44b0-90cc-93ff0139236c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTQ2MzkwMjQsIm5iZiI6MTc1NDYzODcyNCwicGF0aCI6Ii83MzY3ODI0OC80NzU5MTI2MjYtMjYzYTA3YTYtNjk5OS00NGIwLTkwY2MtOTNmZjAxMzkyMzZjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA4MDglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwODA4VDA3Mzg0NFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTVhMmFiMjI1ODIwOGZjYWIyOGJmMjBkMzg1YTYyYTYzNmI3NDJiOTA3NGFmN2NlY2FkOTU1OWUxMTc1NjhlMzMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.ifnUEmBRMX2eTt3dNGX_to8ayAJEgTon1Zv-PrIM3N0">

Run https://github.com/QinYang79/RDE/blob/main/2024-CVPR-RDE/run_rde.sh

You can also check out my training log at https://huggingface.co/Yangsss/RDE

**Comment by li468441680-debug** [comment]
杨博士你好，请问一下您是否在后期尝试过多卡训练

**Comment by QinYang79** [comment]
> 杨博士你好，请问一下您是否在后期尝试过多卡训练

Sorry, there are no plans at the moment. However, based on my experience, converting RDE to multi-GPU training requires attention to the dataloader and how gradients and data pairs are aggregated during CCD and training.
